src/main/java/com/externalSorting/app/ExternalSortingRunner.java

```diff
@@ -9,60 +9,74 @@
     }
 
-    protected void externalSort() {
-
-        System.out.println(outputFolder + "/tmpdir");
-        File newDirectory = new File("src/main/resources/tmpdir/");
-        List<File> files = readAndSplitData("src/main/resources/SmallLibrary.txt", newDirectory);
-
-        System.out.println(outputFolder + "/Output.txt");
-        mergeFiles(files, new File("src/main/resources/SmallLibraryOutput.txt"), comparator);
+    /**
+     * This method is meant to be called from main function.
+     * It is responsible for:
+     *  creating a temporary folder which will hold data files
+     *  calling readAndSplitData and mergeFiles
+     *  deleting temporary folder and files after the process is finished
+     */
+    public void externalSort() {
+        File newDirectory = new File(outputFolder + "/tmpdir");
+        newDirectory.mkdir();
+        List<File> files = readAndSplitData(originalFile, newDirectory);
+        mergeFiles(files, new File(outputFolder + "/Output.txt"));
         for (File file : files) {
             file.delete();
         }
         newDirectory.delete();
     }
+
     /**
-    CHANGE_FILE
-    Maximum number of lines of text that can be put into a single temporary file
+    TMP_FILE_MAXSIZE
+    Maximum number of bytes of a single temporary file. The value of this variable controls memory usage.
     */
-    private static final int TMP_FILE_MAXSIZE = 24000000; // 24mb
+    private static final int TMP_FILE_MAXSIZE = 10000000; // around 10MB
 
     /**
     Defines a default string comparator
     */
-    private static final Comparator<String> comparator = (o1, o2) -> o1.compareTo(o2);
+    private static final Comparator<String> comparator = Comparator.naturalOrder();
 
     /**
-    Reads data from a single text file and splits it into a number of temporary text files
-    containing lists of alphabetically sorted words
-
-    *param* fileName
-    Path to original single text file to be read from
-
-    *param* directory
-    Path of the directory which will contain temporary text files with lists of sorted words
-
-    *return*
-    Returns a list of created temporary files containing sorted data from original file
-    */
-    private static List<File> readAndSplitData(String filePath, File directoryPath) {
+     * Reads data from a single text file and splits it into a number of temporary text files
+     * containing lists of alphabetically sorted words
+     *
+     * @param filePath
+     * Path to the text file from which data will be read
+     * @param directoryPath
+     * Path of the directory which will contain temporary text files with lists of sorted words
+     * @return
+     * Returns a list of created temporary files containing sorted data from original text file
+     */
+    private List<File> readAndSplitData(String filePath, File directoryPath) {
         List<String> buffer = new ArrayList<>();
         List<File> files = new ArrayList<>();
+        int NEWLINE_CHAR_SIZE = 1;
```

Comment:
`NEWLINE_CHAR_SIZE` should be changed to 2.
As per following stackoverflow topic: https://stackoverflow.com/questions/50355675/are-spaces-and-newlines-considered-a-byte-in-java

_The Java programming language represents text in sequences of 16-bit code units, using the UTF-16 encoding.

This means, that each single character (including a new line and a line feed) occupies 16 bit - 2 bytes - in memory._